src/nvim/edit.c

```diff
@@ -2395,44 +2404,41 @@ static void ins_compl_del_pum(void)
   }
 }
 
-/*
- * Return TRUE if the popup menu should be displayed.
- */
-static int pum_wanted(void)
+/// Check if the popup menu should be displayed.
+static bool pum_wanted(void)
+  FUNC_ATTR_PURE FUNC_ATTR_WARN_UNUSED_RESULT
 {
-  /* 'completeopt' must contain "menu" or "menuone" */
-  if (vim_strchr(p_cot, 'm') == NULL)
-    return FALSE;
+  // "completeopt" must contain "menu" or "menuone"
+  if (vim_strchr(p_cot, 'm') == NULL) {
+    return false;
+  }
 
-  /* The display looks bad on a B&W display. */
-  if (t_colors < 8
-      )
-    return FALSE;
-  return TRUE;
+  // The display looks bad on a B&W display.
```

Comment:
weird logic, if the user sets `menu` they do want the menu no matter how bad it looks. But anyway nvim would never initialize `t_colors` below 8 so it shouldn't case any trouble.